### DISCUSSION TITLE
[Documentation]: how to contribute in docs?

### DISCUSSION THREAD

**Discussion by landsman** [storybookjs/storybook#25512] [Help] [answered]
### Describe the problem

Guys, I wasn't able to contribute to the doc, because there are some magic snippets like this one:

```md
<CodeSnippets
  paths={[
    'common/storybook-vite-builder-ts-configure.ts.mdx',
  ]}
/>
```

There is no info on how this folder `common` is created, or where I can find this mdx file.
I tried searching across the project in my IDE and didn't find any mention of this common folder...

Please update [the Contributing manual](https://github.com/storybookjs/storybook/blob/next/CONTRIBUTING.md) ASAP to let us help, this is bad...

### Additional context

I want to add missing types, and info to [Vite builder doc](https://storybook.js.org/docs/react/builders/vite#setup).
Btw why are these snippets in a regular markdown file and it's not mdx as well? It looks like React code.

**Comment by jonniebigodes** [comment, marked as answer]
@landsman, thanks for taking the time to reach out to us with your issue. We appreciate it 🙏 ! To give you a bit of context, as the documentation is housed in the monorepo and the website that houses it is on a different repository, we've decided on this approach. What this means is that once you spin the website, the snippets are interpreted, parsed, and displayed on the website. We're aware of this and planning on improving in the near future. Also, if you look into the monorepo's `docs` folder, you'll see a `snippets` folder. Inside there are all the snippets based on the framework (react, vue, angular) and the `common` folder, which houses every framework-agnostic snippet for the documentation.

If you haven't already, can you please check out [this](https://storybook.js.org/docs/web-components/contribute/new-snippets) section of the documentation? there's a guide on how to contribute to the documentation's snippets.


If you have any questions, let us know, and we'll be more than glad to help you solve your issue. Also, a quick tip for you as you're willing to contribute to the documentation, before pushing the changes into a pull request, please run the following command, `yarn pretty-docs`. Running it will get the snippets properly formatted and aligned with the rest of the documentation. 

Have a great rest of the weekend. 

Stay safe

**Comment by landsman** [comment]
@jonniebigodes thanks for the reply. I'll check it.
